src/components/AddProfileInformation.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  FunctionComponent,
+  KeyboardEvent,
+  useCallback,
+  useEffect,
+  useRef,
+} from "react";
+import { Box } from "@mui/material";
+import { useNavigate } from "react-router-dom";
+import { StringMap } from "common";
+import NEWMLogo from "assets/images/NEWMLogo";
+import { FilledButton, Typography } from "elements";
+import { useFormikContext } from "formik";
+import { FilteredTagsField, GradientTextInputField } from "components";
+
+interface AddProfileInformationProps {
+  readonly fieldName: string;
+  readonly nextRoute: string;
+  readonly prompt: string;
+  readonly helperText?: string;
+  readonly tags?: ReadonlyArray<string>;
+}
+
+const AddProfileInformation: FunctionComponent<AddProfileInformationProps> = ({
+  fieldName,
+  nextRoute,
+  prompt,
+  tags,
+  helperText = "",
+}) => {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const navigate = useNavigate();
+
+  const { errors, handleBlur } = useFormikContext<StringMap>();
+
+  const isValid = !errors[fieldName];
+
+  /**
+   * Navigates if there aren't any
+   * validation errors for the field.
+   */
+  const handleNavigate = () => {
+    if (isValid) {
+      navigate(nextRoute);
+    }
+  };
+
+  const memoizedHandleNavigate = useCallback(handleNavigate, [
+    isValid,
+    navigate,
+    nextRoute,
+  ]);
+
+  useEffect(() => {
+    /**
+     * Formik seems to have issues with triggering the input to blur if
+     * it is focused too early. Delaying the auto focus resolves this.
+     */
+    setTimeout(() => {
+      inputRef.current?.focus();
+    }, 100);
+  }, [inputRef]);
```

Comment:
I was using autofocus originally, but it caused additional issues

Reply:
interesting, i was not aware that was the case, just thought to mention it

Reply:
In retrospect, it's possible the issue I was having was because I wasn't preventing the Formik "enter on submit" functionality from occuring, which may have been what was blurring the form fields and causing the errors to display prematurely. I removed the timeout, and the issue appears to no longer be occuring.

Reply:
Using `autofocus` does causes a linting error, and it may not be optimal for React, so I'll stick with the inputRef focus functionality though.

Reply:
On second thought, the issue looks to still happen in Safari if the timeout isn't present 🤷